### DISCUSSION TITLE
VS Code Error: there is no registered task type 'pnpm'

### DISCUSSION THREAD

**Discussion by fairking** [pnpm/pnpm#5668] [Q&A]
Hi guys,

Maybe it's a stupid question, but I cannot find an answer anywhere including [the documentation](https://pnpm.io/installation).

How do I declare a task in VS Code?

The following throws an error "**Error: there is no registered task type 'pnpm'**" `.vscode/tasks.json`:
```json
{
    "version": "2.0.0",
        "tasks": [
        {
              "label": "pnpm install",
              "type": "pnpm",
              "script": "install",
              "options": {
                "cwd": "${workspaceFolder}/frontend"
              },
              "presentation": {
                "clear": true,
                "panel": "shared",
                "showReuseMessage": false
              },
              "problemMatcher": []
        },
...
```

I tried to install [the following extension](https://marketplace.visualstudio.com/items?itemName=Jacano.vscode-pnpm) but it doesn't work.

**Comment by djsnipa1** [comment]
This should help you...

```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "run pnpm dev",
            "type": "shell",
            "command": "pnpm run dev",
            "options": {
                "cwd": "${workspaceFolder}/frontend"
            },
            "presentation": {
                "clear": true,
                "panel": "shared",
                "showReuseMessage": false
            },
            "problemMatcher": []
        }
    ]
}
```

Notice the key `type`...it will run the `command` in your shell. 

Hope that helps someone!

**Comment by jpc-ae** [comment]
This is a bit of an old one, but unintuitively you want `'type': 'npm'` regardless of what node package manager you're using. If you set the following, VS Code should autodetect your node manager and use the correct one.:

```json
{
	"version": "2.0.0",
	"tasks": [
		{
			"label": "pnpm install",
			"type": "npm",
			"script": "install",
			"options": {
				"cwd": "${workspaceFolder}/frontend"
			},
			"presentation": {
				"clear": true,
				"panel": "shared",
				"showReuseMessage": false
			},
			"problemMatcher": [],
			"detail": "astro build"
		}
	]
}
```

If you want to force it, you can go into settings and search for `pnpm`. The two options are `Npm: Package Manager` and `Npm: Script Runner`. The json settings to force `pnpm` are below:

```json
{
	"npm.packageManager": "pnpm",
	"npm.scriptRunner": "pnpm"
}
```

So far, leaving it on auto has worked fine for me.
